api/peering.go

```diff
@@ -23,53 +35,38 @@ type Peering struct {
 	// Name is the local alias for the peering relationship.
 	Name string
 	// Partition is the local partition connecting to the peer.
-	Partition string `json:"Partition,omitempty"`
+	Partition string `json:",omitempty"`
 	// Meta is a mapping of some string value to any other string value
 	Meta map[string]string `json:",omitempty"`
 	// State is one of the valid PeeringState values to represent the status of
 	// peering relationship.
 	State PeeringState
-	// PeerID is the ID that our peer assigned to this peering.
-	// This ID is to be used when dialing the peer, so that it can know who dialed it.
-	PeerID string
+	// PeerID is the ID that our peer assigned to this peering. This ID is to
+	// be used when dialing the peer, so that it can know who dialed it.
+	PeerID string `json:",omitempty"`
```

Comment:
👀 I recall @kisunji and I chatting about whether these need to be here. I thought only ent specific things like `Partition` or `Namespace` had to have these.

Reply:
When sending data to the consul agent ideally you would only be sending fields with actual data. If you don't include this struct tag then every field is sent do the agent in something like a write request. In the past we've had subtle version incompatibilities caused when the api library became aware of a new field and then that new field was sent to an older consul version, since we hard-fail requests at the consul (json api) agents if they see a field they don't understand.